Sprint 1/step 2

ebedinin opened the pull request.

ebedinin committed:
dev

ebedinin committed:
05.10.2024 office

ebedinin committed:
06.10.2024 home

ebedinin committed:
06.10.2024 home

ebedinin committed:
06.10.2024 home 2

ebedinin committed:
06.10.2024 home 3

ebedinin committed:
06.10.2024 home

ebedinin committed:
07.10.2024 home

ebedinin committed:
08.10.2024 office

ebedinin committed:
08.10.2024 office2

kyzinatra commented on src/App.jsx:
```diff
@@ -0,0 +1,84 @@
+    isError: false,
+    ingridientDetail: null,
+    ingridients: [],
+    order: null
```
Можно лучше: опечатка `ingridients` -> `ingredients`

kyzinatra commented on src/App.jsx:
```diff
@@ -0,0 +1,84 @@
+const modalRoot = document.getElementById("modal");
+function App() {
+  const [state, setState] = React.useState({ 
+    loading: true,
```
Надо исправить: `state` и `setState` ничего не говорят о том, что это за состояния. Стоит придумать более подходящие названия

kyzinatra commented on src/App.jsx:
```diff
@@ -0,0 +1,84 @@
+import React, { useEffect } from 'react';
+import AppHeader from './components/app-header/app-header'
```
Можно лучше: стоит удалить лишние файлы из проекта: тесты, лишние картинки

kyzinatra commented on src/App.jsx:
```diff
@@ -0,0 +1,84 @@
+  })
+  useEffect(()=>{
+    fetch(url)
+    .then(response => response.json())
```
Надо исправить: Стоит добавить проверку успешности выполнения с помощью `res.ok` так как `catch` ловит только ошибки в самом запросе, но если сервер вернет к примеру 301, то ее `catch` не поймает.

[Подробнее:](https://developer.mozilla.org/ru/docs/Web/API/Fetch_API/Using_Fetch) (https://developer.mozilla.org/ru/docs/Web/API/Fetch_API/Using_Fetch)

kyzinatra commented on src/App.jsx:
```diff
@@ -0,0 +1,84 @@
+  },[])
+  const buns = state.ingridients.filter((item)=>item.type==="bun");
+  const sauces = state.ingridients.filter((item)=>item.type==="sauce");
+  const main = state.ingridients.filter((item)=>item.type==="main");
```
Можно лучше: фильтры лучше обернуть в `useMemo()`

kyzinatra commented on src/App.jsx:
```diff
@@ -0,0 +1,84 @@
+    .catch(()=> setState({...state,isError:true}))
+  },[])
+  const buns = state.ingridients.filter((item)=>item.type==="bun");
+  const sauces = state.ingridients.filter((item)=>item.type==="sauce");
```
Можно лучше: кажется, что фильтрация ингредиентов должна быть ответственностью `BurgerIngredients`, как и вывод `showIngridient`. Так что лучше перенести эту логику туда

kyzinatra commented on src/App.jsx:
```diff
@@ -0,0 +1,84 @@
+    loading: true,
+    isError: false,
+    ingridientDetail: null,
+    ingridients: [],
```
Можно лучше: не стоит хранить все состояние в 1 `state`. Вы можете разбить эти состояния и хранить все там, где вам нужно

kyzinatra commented on src/App.jsx:
```diff
@@ -0,0 +1,84 @@
+}
+const closeOrder = (e)=>{
+setState({
+...state,order:null
```
Можно лучше: рекомендую установить `prettier` для автоматического форматирования кода

kyzinatra commented on src/App.jsx:
```diff
@@ -0,0 +1,84 @@
+        </Modal>
+    }
+    {state.order && 
+      <Modal root={modalRoot} title={''} close={closeOrder}>
```
Можно лучше: модальные окна стоит размещать там, где они нужны, а не в корне проекта. Так код получится более чистым. А порталы все равно отрендерят их там, где надо

kyzinatra commented on src/components/modal/modal-overlay/modal-overlay.jsx:
```diff
@@ -0,0 +1,26 @@
+
+    const clickModalOverlay = React.useCallback((e)=>{
+        if (e.target.className === style.modalOverlay) {
+           props.close()
```
Можно лучше: тут `useCallback` можно не использовать, так как мемоизировать функцию тут не нужно

kyzinatra commented on src/components/modal/modal.jsx:
```diff
@@ -0,0 +1,42 @@
+    React.useEffect(()=>{
+        document.addEventListener("keydown", pressEsc, false);
+        return ()=> document.removeEventListener("keydown", pressEsc, false);
+    },[])
```
Отлично! Круто, что вы очищаете обработчик

kyzinatra commented on src/components/modal/modal.jsx:
```diff
@@ -0,0 +1,42 @@
+    
+    const pressEsc = React.useCallback((e)=>{
+        if (e.key === "Escape") {
+           props.close()
```
Можно лучше: чтобы не использовать `useCallback` функцию `pressEsc` можно перенести внутрь `useEffect`

kyzinatra commented on src/components/modal/modal.jsx:
```diff
@@ -0,0 +1,42 @@
+    close: PropTypes.func,
+    title:PropTypes.string,
+    root: PropTypes.object,
+    children: PropTypes.object
```
Надо исправить: чтобы типизировать `react node` можно использовать `.node`

kyzinatra commented on src/components/burger-constructor/burger-constructor.jsx:
```diff
@@ -0,0 +1,65 @@
+    createOrder: PropTypes.func
+}
+
+export {BurgerConstructor};
```
Можно лучше: стоит добавить `.isRequired` везде

kyzinatra commented:
Выглядит очень круто! Видно, что проделана большая работа. Но все же есть моменты на которые стоит обратить внимание.
Подробнее: 

**Что важно исправить:**

-  `state` и `setState` ничего не говорят о том, что это за состояния. Стоит придумать более подходящие названия
-  Стоит добавить проверку успешности выполнения с помощью `res.ok` так как `catch` ловит только ошибки в самом запросе, но если сервер вернет к примеру 301, то ее `catch` не поймает.

[Подробнее:](https://developer.mozilla.org/ru/docs/Web/API/Fetch_API/Using_Fetch) (https://developer.mozilla.org/ru/docs/Web/API/Fetch_API/Using_Fetch)
-  чтобы типизировать `react node` можно использовать `.node`


**Можно лучше:**

-  опечатка `ingridients` -> `ingredients`
-  стоит удалить лишние файлы из проекта: тесты, лишние картинки
-  фильтры лучше обернуть в `useMemo()`
-  кажется, что фильтрация ингредиентов должна быть ответственностью `BurgerIngredients`, как и вывод `showIngridient`. Так что лучше перенести эту логику туда
-  не стоит хранить все состояние в 1 `state`. Вы можете разбить эти состояния и хранить все там, где вам нужно
-  рекомендую установить `prettier` для автоматического форматирования кода
-  модальные окна стоит размещать там, где они нужны, а не в корне проекта. Так код получится более чистым. А порталы все равно отрендерят их там, где надо
-  тут `useCallback` можно не использовать, так как мемоизировать функцию тут не нужно
-  чтобы не использовать `useCallback` функцию `pressEsc` можно перенести внутрь `useEffect`
-  стоит добавить `.isRequired` везде


**Что понравилось:**

-  Круто, что вы очищаете обработчик


**Обратите внимание, что работа принимается только после исправления всех `«Надо исправить»`.**

ebedinin committed:
review №1

1)state и setState ничего не говорят о том, что это за состояния.
Стоит придумать более подходящие названия
(https://github.com/ebedinin/react-burger/pull/1#discussion_r1792055116)
2)Стоит добавить проверку успешности выполнения с помощью res.ok так как
catch ловит только ошибки в самом запросе, но если сервер вернет к
примеру 301, то ее catch не поймает.
(https://github.com/ebedinin/react-burger/pull/1#discussion_r1792056382)
3)тобы типизировать react node можно использовать .node
(https://github.com/ebedinin/react-burger/pull/1#discussion_r1792064612)

kyzinatra commented on src/App.jsx:
```diff
@@ -10,71 +10,82 @@ import {Modal} from './components/modal/modal.jsx'
-    .catch(()=> setState({...state,isError:true}))
+    .then(response => {
+      if (response.status!==200){
+        setIngridients({...ingridients,isError:true})
```
Надо исправить: нужно проверять либо `res.ok` либо все 2хх статусы, так как к примеру 203 тоже успех

kyzinatra commented:
Вы хорошо постарались, но для того, чотбы окончательно принять работу необходимо доисправить некоторые моменты
 - нужно проверять либо `res.ok` либо все 2хх статусы, так как к примеру 203 тоже успех
